bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.energenie.internal.handler;
+
+import static org.openhab.binding.energenie.internal.EnergenieBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.nio.charset.Charset;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergeniePWMHandler} is responsible for reading states and update PWM channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+
+@NonNullByDefault
+public class EnergeniePWMHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private String host = "";
+    private String password = "";
+
+    /** the timeout to use for connecting to a given host (defaults to 5000 milliseconds) */
+    private int timeout = 6000;
+
+    /**
+     * The default refresh interval in Seconds.
+     */
+    private int DEFAULT_REFRESH_INTERVAL = 60;
+    @Nullable
+    private ScheduledFuture<?> refreshJob;
+    private Runnable refreshRunnable = new Runnable() {

```

Comment:
Remove `refreshRunnable` and replace with `this:getState`.

Reply:
Don't Understand your suggestion, could you pleas explain.

Reply:
You don't need the variable `refreshRunnable`, but instead of passing `refreshRunnable` you can pass `this::getState` to the scheduler.

Reply:
Now I got you, there was a missing ":" in your first comment, causing errors in Eclipse ....

Reply:
Done